### PR TITLE
Canvas initializer

### PR DESCRIPTION
**Pull request by CarsonFox**
Includes a dead simple install script and some extra info in the readme

### CODE REVIEW

File: post_events.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# This code mostly comes from config/initializers/live_events.rb
+
+class CustomEventClient
+  def self.put_records(records:, stream_name:)
+    events = records.map do |e|
+      data = JSON.parse(e[:data])
+      data['metadata'] = data.delete('attributes')
```
Comment (CarsonFox) [inline post_events.rb:7]: The documentation says this field should be named "metadata", but my local canvas instance uses "attributes". I'd rather match the documentation and just rename the key here.

File: bin/install_initializer.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+cp post_events.rb "$1/config/initializers/"
```
Comment (CarsonFox) [inline bin/install_initializer.sh:7]: Should this be a symlink?

**Review by CarsonFox — COMMENTED** [review]
I've added an event name field to the jwt sent by the canvas initializer. This means we won't have to parse the payload json once to get the event name.

File: controllers/events.go [hunk 3]
```diff
@@ -11,29 +10,35 @@ func (c *ControllerContext) NewEventsHandler() http.HandlerFunc {
...
 		switch r.Method {
 		case "POST":
-			jwtData := r.Context().Value(middleware.DecodedJwtKey).(map[string]interface{})
-			eventsString := jwtData["events"].(string)
-			eventPayloads, err := lib.EventPayloadsFrom([]byte(eventsString))
-			if err != nil {
-				panic(err)
-			}
+			jwtData := middleware.GetJwtClaims(r)
+			events := jwtData["events"].([]interface{})
```
Comment (sethduffin) [inline controllers/events.go:14]: Nice, this is way better. We should remove the `EventPayloadsFrom` method from the project, I don't see where else we'd actually use that.

File: controllers/events.go [hunk 4]
```diff
@@ -11,29 +10,35 @@ func (c *ControllerContext) NewEventsHandler() http.HandlerFunc {
...
-					panic(err)
+				eventName, ok := eventMap["name"]
+				if !ok {
+					panic(fmt.Errorf("missing event name in payload"))
 				}
 
 				switch eventName {
 				case "discussion_entry_created":
-					err = c.Repo.DiscussionEntryCreatedEvent.CreateFrom(payload)
+					payload, ok := eventMap["payload"]
```
Comment (sethduffin) [inline controllers/events.go:29]: Looks good! Let's update atomic-event-mocks to match the JWT claims expected here (`name` & `payload`).

**Review by sethduffin — APPROVED** [review]

File: bin/install_initializer.sh (hunk 2 above)
Comment (kylehovey) [reply bin/install_initializer.sh:7]: I think so

**Review by kylehovey — APPROVED** [review]

**Pull request merged**